pkgs/applications/editors/vscode-with-extensions/default.nix

```diff
@@ -0,0 +1,107 @@
+/*
+  `vsixExtensions`
+   :  A set of vscode extensions to be installed alongside the editor. Here's a an
+      example:
+
+      ~~~
+      vscode-with-extensions.override {
+        vsixExtensions = {
+          "nix" = {
+            publisher = "bbenoist";
+            version = "1.0.1";
+            sha256 = "0zd0n9f5z1f0ckzfjr38xw2zzmcxg1gjrava7yahg5cvdcw6l35b";
+          };
```

Comment:
I would create a new namespace like: `vscode-extensions` similar to `./pkgs/misc/vim-plugins/default.nix`,
so the user only has to pass a list:

```
vscode-with-extensions.override {
  vsixExtensions = with vscode-extensions; [ nix ];
}
```

then I would also create an abstraction similar to `buildVimPluginFrom2Nix` for the following reason: 
I suppose that some plugins require patching to find the right external tools.